src/nvim/tui/tui.c

```diff
@@ -923,19 +925,26 @@ static void tui_mouse_off(UI *ui)
 
 static void tui_set_mode(UI *ui, ModeShape mode)
 {
-  if (!cursor_style_enabled) {
+  if (!cursor_style_enabled || !ui->rgb) {
     return;
   }
   TUIData *data = ui->data;
   cursorentry_T c = data->cursor_shapes[mode];
 
-  if (c.id != 0 && ui->rgb) {
+  if (c.id != 0) {
     int attr = syn_id2attr(c.id);
     if (attr > 0) {
       HlAttrs *aep = syn_cterm_attr2entry(attr);
-      UNIBI_SET_NUM_VAR(data->params[0], aep->rgb_bg_color);
-      unibi_out_ext(ui, data->unibi_ext.set_cursor_color);
+      int attr = ui->rgb ? aep->rgb_ae_attr : aep->cterm_ae_attr;
```

Comment:
Line 928 returns early if `!ui->rgb`, so I guess this is in preparation for the future when we support cursor-coloring for non-rgb UIs?

Reply:
Yes. I guess it is a little bit redundant.  I will remove it.